fiftyone/operators/builtin.py

```diff
@@ -24,46 +24,35 @@ def config(self):
 
     def resolve_input(self, ctx):
         inputs = types.Object()
-        sample_ids = ctx.selected
-        count = len(sample_ids)
-        header = "Clone sample"
+
+        count = len(ctx.selected)
         if count > 0:
             sample_text = "sample" if count == 1 else "samples"
-            header = f"Clone {count} {sample_text}?"
             inputs.str(
                 "msg",
-                label=f"Press 'Execute' to clone {count} selected {sample_text}",
-                view=types.Notice(space=6),
-            )
-            inputs.str(
-                "btn",
-                label="Show selected samples",
-                view=types.Button(operator="show_selected_samples", space=3),
+                label=f"Clone {count} selected {sample_text}?",
+                view=types.Warning(),
             )
         else:
-            header = "No selected samples"
             inputs.str(
                 "msg",
                 label="You must select samples in the grid to clone",
                 view=types.Warning(),
             )
 
-        return types.Property(inputs, view=types.View(label=header))
+        view = types.View(label="Clone selected samples")
+        return types.Property(inputs, view=view)
 
     def execute(self, ctx):
-        sample_ids = ctx.selected
-        if len(sample_ids) > 0:
-            samples = ctx.dataset.select(sample_ids)
-            cloned_samples = [
-                fo.Sample.from_dict(dict(**sample.to_dict(), id=None))
-                for sample in samples
-            ]
-            ctx.dataset.add_samples(cloned_samples)
-            ctx.trigger("reload_samples")
-            ctx.trigger(
-                "show_samples",
-                {"samples": [sample.id for sample in cloned_samples]},
-            )
+        if not ctx.selected:
+            return
+
+        samples = ctx.dataset.select(ctx.selected)
+        sample_ids = ctx.dataset.add_collection(samples, new_ids=True)
+
+        ctx.trigger("clear_selected_samples")
+        ctx.trigger("reload_samples")
+        ctx.trigger("show_samples", {"samples": sample_ids})
```

Comment:
@ritch or @imanjra neither of these operations seem to work:

```py
ctx.trigger("show_samples", {"samples": sample_ids})
ctx.trigger("set_selected_samples", {"samples": sample_ids})
```

Can you investigate what's wrong with them?

Reply:
`show_samples` raises this error:

```
string indices must be integers
```

And if I try to switch to `set_selected_samples`, I get this error:

```
TypeError: Cannot read properties of undefined (reading 'toLocaleString')
Trace
TypeError: Cannot read properties of undefined (reading 'toLocaleString')
    at Evt (http://localhost:5151/assets/index.e75ca0f1.js:5142:2058)
    at sye (http://localhost:5151/assets/index.e75ca0f1.js:87:19225)
    at tde (http://localhost:5151/assets/index.e75ca0f1.js:89:12599)
    at $6e (http://localhost:5151/assets/index.e75ca0f1.js:89:45831)
    at j6e (http://localhost:5151/assets/index.e75ca0f1.js:89:40577)
    at iwt (http://localhost:5151/assets/index.e75ca0f1.js:89:40500)
    at jz (http://localhost:5151/assets/index.e75ca0f1.js:89:40349)
    at cde (http://localhost:5151/assets/index.e75ca0f1.js:89:36649)
    at xRe (http://localhost:5151/assets/index.e75ca0f1.js:89:37468)
    at nT (http://localhost:5151/assets/index.e75ca0f1.js:87:3316)
```

Reply:
Note that `ctx.trigger("show_samples", {"samples": sample_ids})` is already used by the current `clone_selected_samples` operator on `develop`; so there's a bug independent of this PR

Reply:
Interesting. It's likely an existing bug. Updating the view flow has changed a bit recently. Taking a look

Reply:
`ctx.trigger("reload_samples")` triggers a hard reload so I believe any triggers after that would not work

Reply:
PR https://github.com/voxel51/fiftyone/pull/3818 going into this PR should fix both built-in operators

Reply:
@imanjra thank you! merged ✔️